### DISCUSSION TITLE
How do chunks become viewable?

### DISCUSSION THREAD

**Discussion by Moulberry** [Minestom/Minestom#606] [Help] [answered]
It seems as though chunk#sendPacketToViewers isn't sending the packet even though I can see the chunk
Is this a bug?

**Comment by TheMode** [comment, marked as answer]
Chunks currently get their viewers from the instance `EntityTracker`: https://github.com/Minestom/Minestom/blob/master/src/main/java/net/minestom/server/instance/Chunk.java#L71-L72

`updateTracker` will retrieve lists that contain the players in surrounding chunks, updated in real-time. It is this way to avoid manipulating chunk specific collections when crossing border.
